### ISSUE TITLE
Orchestrating Robot Framework with Celery - issue when gracefully cancelling Robot Framework tests

### ISSUE THREAD

**Issue by sokol1412** [robotframework/robotframework#5066]
Hi!

I have specific use case, where I'm executing Robot Framework tests through robot.run(....) method, and that method call is a part of Celery task which is responsible for setting up, executing and further reporting of the tests.

I'm now trying to implement a functionality of canceling the specific Celery task, which would also result in graceful termination of underyling Robot Framework execution (so that Robot report is generated etc.)

`celery_app.control.revoke(celery_task_id_to_cancel, terminate=True, signal="SIGINT")`

With above line I can achieve exactly what I need -> it gracefully terminates Robot Framework tests, generates reports and then sets celery task status to Revoked.
![image](https://github.com/robotframework/robotframework/assets/17600308/eb06baec-6351-48b8-b022-9a539743fe32)

What is surprising and unexpected, after that I'm spawning completely new celery task and underyling Robot Framework execution is not able to start, it fails right when it's started (and I have not canceled this new task):

![image](https://github.com/robotframework/robotframework/assets/17600308/e94777be-0886-420f-b240-cd3385ffb9ed)

Important to mention, the above behavior is not 100% reproducible. Sometimes after triggering second task (when first one was canceled) it runs ok and Robot Framework tests are able to complete.

Only after restarting the worker container (Celery worker is running on Docker container), above issue is fixed, but then after first Cancel the issues start to appear again.

I suspect it's some threading collision between Celery and Robot Framework, maybe you have any ideas why something like that could happen? Thanks!

**Comment by pekkaklarck** [comment]
Looks pretty odd. Are you running Robot always as a new process or do you run it programmatically using same Python instance? In the latter case it's possible that Robot's internal state is messed up after graceful termination and new execution doesn't work.

**Comment by Bouska** [comment]
I have either this issue or a similar issue (but for now it sort of a feature for me). I'm running the following in a loop in Python, changing the variable file in each iteration:
```
run_cli(
                [
                    "--outputdir",
                    output_dir,
                    "--variablefile",
                    f"{FILE_DIR}/confs/{conf_id}.json",
                    f"{FILE_DIR}/test_procedure_1.robot",
                ],
                exit=False,
            )
```
If I send a SIGINT (with a KeyboardInterrupt in the console running the Python script), all the call tests from all the the `run_cli` calls are failed with `Test execution stopped due to a fatal error.`

**Comment by pekkaklarck** [comment]
@Bouska Signals stopping execution is by design. You probably can avoid that by registering a custom signal handler.

There hasn't been updates to the original issue so I close this.
